scripts/release/version.py

```diff
@@ -0,0 +1,50 @@
+import semver
+from git import Commit, Repo, TagReference
+
+from scripts.release.changelog import ChangeKind
+
+
+def find_previous_version(repo: Repo, initial_commit_sha: str = None) -> (TagReference | None, Commit):
+    """Find the most recent version that is an ancestor of the current HEAD commit."""
```

Comment:
q: I believe you said that the tool supports multiple projects in the same repo. How is it going to work? Do they all need to share the versioning?

Or have I misunderstood something?